### PR TITLE
fix(useStorage): default null values will not be stored in local storage

### LINKED ISSUE #597: [bug] useStorage default value `null` get's written to localstorage and read as `"null"`
If you set `null` as the default value of `useStorage`, this value get's written to the localStorage on initial execution and is then read as the string `"null"`. This behaviour was most likely introduced with #528 and I suspect it isn't intended. 
All versions >=4.11.1 are affected.

Example:
```
import { useStorage } from "@vueuse/core";
const test = useStorage("key", null);
console.log("Key:", typeof test.value, test.value);

// First load (expected behaviour)
// > Key: object null
// Subsequent loads (broken)
// > Key: string "null"
```

### PR DESCRIPTION
**Pull request by jeffrey-dot-li**
`useStorage('key', null)` now no longer puts `{'key' : 'null'}` into local storage. All other useStorage behaviour is now working as expected: 
`state.value = null` now always deletes the item from local storage and sets `state.value` to be `null` as expected.
If the value of 'null' *is* stored within local storage, it is no longer treated any differently than any other string.
`useStorage('key', null)` now works as the default value for `useStorage` without any other strange behaviour.

fixes #597 properly

### CODE REVIEW

File: packages/core/useStorage/index.ts [hunk 1]
```diff
@@ -11,23 +11,23 @@ export type Serializer<T> = {
...
   },
   any: {
-    read: (v: any) => (v != null && v !== 'null') ? v : null,
+    read: (v: any) => v,
     write: (v: any) => String(v),
   },
   string: {
-    read: (v: any) => v != null ? v : null,
+    read: (v: any) => v,
     write: (v: any) => String(v),
```
Comment (adamdehaven) [inline packages/core/useStorage/index.ts:31]: @jeffrey-dot-li if the user passes `null` wouldn't this still convert it to a string of `"null"`? or am I missing something?

Comment (jeffrey-dot-li) [reply packages/core/useStorage/index.ts:31]: Nah if the use passes `null` it doesn't call `serializer.write` in the first place so there is no string conversion - line 115:
```
        if (defaultValue !== null)
          storage.setItem(key, serializer.write(defaultValue))
```

**Comment by antfu** [comment]
Can you target to the `next` branch instead. While I believe this is the correct way to do, but there might be some users already rely on this behavior, so we'd better to release it as a breaking change. Thanks!

**Comment by adamdehaven** [comment]
> Can you target to the `next` branch instead. While I believe this is the correct way to do, but there might be some users already rely on this behavior, so we'd better to release it as a breaking change. Thanks!

@antfu any idea when the next branch will be released? I'm using this in a big MVP and would be a big help.

_Also, I wasn't able to test locally so I'm depending on someone else verifying this is the fix 😃_

**Comment by antfu** [comment]
I release it quite frequently as beta, you can see the releases tab in the repo.

**Pull request merged**